### ISSUE TITLE
Propagate an error message when a tx already exists in the mempool

### ISSUE THREAD

**Issue by rootulp** [celestiaorg/celestia-core#964]
## Context

https://github.com/rollkit/rollkit/issues/725

## Problem

When clients attempt to PayForData in celestia-app, celestia-core doesn't return a full error message when a tx already exists in the mempool. The `celestia-appd` binary will log:

```log
6:26PM ERR Error on broadcastTxCommit err="tx already exists in cache" module=rpc
```

but celestia-core returns to celestia-app something like

```
[rootulp] txResp: tx_response:<txhash:"7AE1E7393D312FFA9E9965DD5958FE83E0A741A0F3345B68A2F7F6039EECB396" codespace:"sdk" code:19 > , err <nil>

[rootulp] core_access.go SubmitPayForData response: code: 19
codespace: sdk
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: ""
timestamp: ""
tx: null
txhash: 7AE1E7393D312FFA9E9965DD5958FE83E0A741A0F3345B68A2F7F6039EECB396
, err: <nil>
```

which means celestia-app can't forward a full error message to celestia-node or rollkit.

## Proposal

Modify the txResponse to include a descriptive error message or return an error. Prior to making any changes, investigate if the relevant code path has been modified from vanilla tendermint/tendermint.

**Comment by evan-forbes** [comment]
responding with more human readable errors is definitely something that would be nice, and perhaps we can do something further with the cli, but we are returning the error code 19

https://github.com/celestiaorg/cosmos-sdk/blob/0aef7ba8d215a3978de8c298dc40d9fa65041695/types/errors/errors.go#L95

typically, when checking for responses after submitting txs, we're supposed to check this code. if the code is not 0, then something went wrong

**Comment by rootulp** [comment]
> but are we checking the abci code after submitting a tx

Which component do you expect to check the ABCI code? For other types of transaction errors, an appropriate error message is bubbled up all the way up to Ethermint. For example:

```error
7:13PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 4, Message: signature verification failed; please verify account number (0) and chain-id (): unauthorized" attempt=9 module=BlockManager
```

```error
6:45PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 13, Message: insufficient fees; got:  required: 6000utia: insufficient fee" attempt=1 module=BlockManager
```

so I'm concerned the "tx already exists in mempool" error is handled differently in celestia-core which leads to a lack of error message.

**Comment by evan-forbes** [comment]
> Which component do you expect to check the ABCI code?

if its json, then the `code` field, but if grpc, then we should get 
https://github.com/celestiaorg/celestia-core/blob/07b8dbcb153563db8f2429b94a632b23d2621519/proto/tendermint/abci/types.proto#L241

an example would be 
https://github.com/celestiaorg/celestia-app/blob/a113f5ac8ad5f336f419d47cf55e0510df550490/app/test/integration_test.go#L167-L169

> so I'm concerned the "tx already exists in mempool" error is handled differently in celestia-core which leads to a lack of error message

I see, that makes sense

**Comment by evan-forbes** [comment]
I'm assuming that at least the code is returned back because of
> [rootulp] core_access.go SubmitPayForData response: **code: 19**

**Comment by rootulp** [comment]
Yep the Cosmos SDK error code is returned in the tx response so it is possible to wrap the tx response with an error message in celestia-app somewhere after [this line](https://github.com/celestiaorg/celestia-app/blob/a138c19a2215747e11c48bd3ef810bbca29bd93a/x/blob/payforblob.go#L49) (i.e. with a switch statement using the Cosmos SDK errors [here](https://github.com/celestiaorg/cosmos-sdk/blob/0aef7ba8d215a3978de8c298dc40d9fa65041695/types/errors/errors.go#L95) but that seems less ideal than having celestia-core return the error message along with the error code.

**Comment by evan-forbes** [comment]
we might want to upstream this, and it sounds like this is only due this specific error in the mempool

**Comment by evan-forbes** [comment]
can close since status is indexed now
